src/services/authentication/strategies/TSPStrategy.js

```diff
@@ -0,0 +1,97 @@
+const { AuthenticationBaseStrategy } = require('@feathersjs/authentication');
+const { NotAuthenticated } = require('@feathersjs/errors');
+
+const logger = require('../../../logger');
+const {
+	decryptToken, verifyToken,
+	ENTITY_SOURCE, SOURCE_ID_ATTRIBUTE,
+} = require('../../sync/strategies/TSP/TSP');
+const { SYNCER_TARGET } = require('../../sync/strategies/TSP/TSPSchoolSyncer');
+
+class TSPStrategy extends AuthenticationBaseStrategy {
+	get configuration() {
+		const authConfig = this.authentication.configuration;
+		const config = super.configuration || {};
+		return {
+			service: authConfig.service,
+			entity: authConfig.entity,
+			entityId: authConfig.entityId,
+			errorMessage: 'Invalid login',
+			entityUsernameField: config.usernameField,
+			...config,
+		};
+	}
+
+	verifyConfiguration() {}
+
+	async getPayload(authResult, params) {
+		// Call original `getPayload` first
+		const payload = await super.getPayload(authResult, params);
+		const { user } = authResult;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['user' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/schul-cloud/schulcloud-server/pullRequest?prid=4605061)